From 3b7bd0b7a7f0f5f1bf7609bdf34e7d9a32608ea6 Mon Sep 17 00:00:00 2001
From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Tue, 17 Mar 2015 11:50:06 +0100
Subject: [PATCH] bugfix: size-based legacy config statements did not work
 properly

on some platforms they misadressed memory, which could also lead
to a segfault on startup. The problem is NOT experience on amd686
in 64bit builds -- that's probably the reason this bug was uncovered
very late. We assume, it's present in all v8 versions.

Thanks to Michael Biebl for alerting us of it.

closes https://github.com/rsyslog/rsyslog/issues/270
---
 runtime/cfsysline.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/cfsysline.c b/runtime/cfsysline.c
index 36d4679..b441a48 100644
--- a/runtime/cfsysline.c
+++ b/runtime/cfsysline.c
@@ -161,7 +161,7 @@ static rsRetVal parseIntVal(uchar **pp, int64 *pVal)
  * param value must be int64!
  * rgerhards, 2008-01-09
  */
-static rsRetVal doGetSize(uchar **pp, rsRetVal (*pSetHdlr)(void*, uid_t), void *pVal)
+static rsRetVal doGetSize(uchar **pp, rsRetVal (*pSetHdlr)(void*, int64), void *pVal)
 {
 	DEFiRet;
 	int64 i;
